### PR TITLE
Move dataset_config/ from api/ to root

### PR DESCRIPTION
**Pull request by melissachang**
To be consistent with data-explorer-indexers repo.

Tested:
- Local default dataset
- Local Platinum Genomes
- deploy/deploy-api.sh Platinum Genomes

### CODE REVIEW

**Comment by melissachang** [comment]
Done, I had to fix .gitignore

File: dataset_config/test_dataset/dataset.json [hunk 1]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (alahwa) [inline dataset_config/test_dataset/dataset.json]: nit: intentional whitespace here?

File: dataset_config/test_dataset/ui.json [hunk 2]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (alahwa) [inline dataset_config/test_dataset/ui.json]: And here?

File: dataset_config/test_dataset/dataset.json (hunk 1 above)
Comment (melissachang) [reply dataset_config/test_dataset/dataset.json]: Done

File: dataset_config/test_dataset/ui.json (hunk 2 above)
Comment (melissachang) [reply dataset_config/test_dataset/ui.json]: Done

**Pull request merged**